### PR TITLE
add support for retrieving the last N versions of a secret

### PR DESCRIPTION
**Pull request by jfromaniello**
Credstash support [versioning of secrets](https://github.com/fugue/credstash#versioning-secrets) which allows to easily rotate secrets.

With this change node-credstash will keep its current behavior of returning the latest version of a secret by default.

I added a new option `limit`, to get the latest N versions of a secret:

``` js
const Credstash = require('../index.js');

var credstash = new Credstash();
return credstash.get('secret', { limit: 3 }, (e, secrets) => {
  console.log('this is the last version', secrets[0]);
  console.log('this is the second-last', secrets[1]);
  console.log('this is the third-last', secrets[2]);
});
```

If `options.limit: 1` or if you don't pass the options parameters, the callback receives `secret` (singular). So, this change is backward compatible.

### CODE REVIEW

**Comment by jfromaniello** [comment]
There are two things that I am not entirely convinced:
1. The name of the parameter `limitd` - Any idea for a better name?
2. The callback returning `secret` or `secrets`.. Maybe it should return `secrets` always, even when retrieving just 1, and release this change as a major version.

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
   "dependencies": {
     "aes-js": "0.2.2",
     "async": "1.5.2",
-    "aws-sdk": "2.2.35"
+    "aws-sdk": "2.2.35",
+    "xtend": "~4.0.1"
```
Comment (roylines) [inline package.json]: I've tended to hardcode actual versions. Could you please resubmit without the ~

File: .jshintrc [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
```
Comment (roylines) [inline .jshintrc]: Could you please remove this file from the pull request.

**Comment by roylines** [comment]
Hi Jose,

Firstly thanks for the pull request! Just made a couple of comments above. If you could fix the versioning of xtend and remove the .jshintrc file I'm happy to merge and republish.

I'm ok with the name limit, and I think it would be better to keep backwards compatibility for now.

**Comment by jfromaniello** [comment]
Done! thanks a lot

**Comment by jfromaniello** [comment]
Thank you for merging this, is this released already?

**Comment by roylines** [comment]
Yes it is

**Pull request merged**